src/progress-steps/styled-components.js

```diff
@@ -55,6 +56,12 @@ export const StyledIcon = styled<StylePropsT>(
 
     const marginTop = `calc((${font.lineHeight} - ${size}) / 2)`;
 
+    if ($theme.direction === 'rtl') {
+      let oldMarginLeft = marginLeft;
```

Comment:
`[marginLeft, marginRight] = [marginRight, marginLeft]` would do the trick too - feel free to disregard